### PR TITLE
:rocket: Change Hub's and Front's resource type from `Pod` to `Deployment`

### PR DESCRIPTION
**Pull request by Ziul**

### CODE REVIEW

File: helm-chart/templates/04-hub-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            - ./hub
+            {{ .Values.tap.debug | ternary "- -debug" "" }}
+          envFrom:
+            - configMapRef:
+                name: kubeshark-hub-config
+            - secretRef:
+                name: kubeshark-hub-secret
+          env:
+            - name: POD_REGEX
+              value: '{{ .Values.tap.regex }}'
```
Comment (mertyildiran) [inline helm-chart/templates/04-hub-deployment.yaml:38]:
```suggestion
```

Duplicate. Already defined in `kubeshark-hub-config`.

File: kubernetes/provider.go [hunk 2]
```diff
@@ -127,6 +127,30 @@ func (provider *Provider) ListAllRunningPodsMatchingRegex(ctx context.Context, r
...
+
+func (provider *Provider) ListAllNamespaces(ctx context.Context) ([]core.Namespace, error) {
+	namespaces, err := provider.clientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return namespaces.Items, err
+}
+
```
Comment (mertyildiran) [inline kubernetes/provider.go:153]:
```suggestion
```

Dead code.

File: kubernetes/proxy.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ const selfServicePort = 80
 
 func StartProxy(kubernetesProvider *Provider, proxyHost string, srcPort uint16, selfNamespace string, selfServiceName string) (*http.Server, error) {
 	log.Info().
+		Str("proxyHost", proxyHost).
```
Comment (mertyildiran) [inline kubernetes/proxy.go:27]:
```suggestion
		Str("proxy-host", proxyHost).
```

We use `kebab-case` style for debugging values in `log` calls project-wide.

**Review by mertyildiran — CHANGES_REQUESTED** [review]

File: helm-chart/templates/04-hub-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+apiVersion: apps/v1
+kind: Deployment
```
Comment (mertyildiran) [inline helm-chart/templates/04-hub-deployment.yaml:2]:
```suggestion
kind: StatefulSet
```

Hub should be `StatefulSet`.

**Review by mertyildiran — CHANGES_REQUESTED** [review]

File: helm-chart/templates/04-hub-deployment.yaml (hunk 4 above)
Comment (Ziul) [reply helm-chart/templates/04-hub-deployment.yaml:2]: From what I saw its uses, it doesn't need to be StatefulSet. Workers can easily find it, as it keeps watching.

Comment (mertyildiran) [reply helm-chart/templates/04-hub-deployment.yaml:2]: OK, let's keep it as `Deployment`.

**Review by mertyildiran — APPROVED** [review]

**Pull request merged**